examples/efficientdet/efficientdet_building_blocks.py

```diff
@@ -0,0 +1,321 @@
+            for level in range(min_level, max_level + 1):
+                for i, fnode in enumerate(reversed(self.fpn_config['nodes'])):
+                    if fnode['feat_level'] == level:
+                        feats.append(cell_feats[-1 - i])
+                        break
+        return feats
+
+
+class FPNCell(tf.keras.layers.Layer):
+    """A single FPN cell."""
+    def __init__(self, config, name='fpn_cell'):
+        super().__init__(name=name)
+        self.config = config
+        self.fpn_config = get_fpn_configuration(config['fpn_name'],
+                                                config['min_level'],
+                                                config['max_level'],
+                                                config['fpn_weight_method']
+                                                )
+        self.fnodes = []
+        for i, fnode_cfg in enumerate(self.fpn_config['nodes']):
+            fnode = FNode(
+                fnode_cfg['feat_level'] - self.config['min_level'],
+                fnode_cfg['inputs_offsets'],
+                config['fpn_num_filters'],
+                config['apply_bn_for_resampling'],
+                config['conv_after_downsample'],
+                config['conv_bn_act_pattern'],
+                config['separable_conv'],
+                config['act_type'],
+                weight_method=self.config['fpn_weight_method'],
+                data_format=self.config['data_format'],
+                name='fnode%d' % i)
+            self.fnodes.append(fnode)
+
+    def call(self, feats, training):
+
+        for fnode in self.fnodes:
+            feats = fnode(feats, training)
+        return feats
+
+
+def add_n(nodes):
+    """A customized add_n to add up a list of tensors."""
+    with tf.name_scope('add_n'):
+        new_node = nodes[0]
+        for n in nodes[1:]:
+            new_node = new_node + n
+        return new_node
+
+
+class FNode(tf.keras.layers.Layer):
+    """A keras layer implementing BiFPN node."""
+
+    def __init__(self,
+                 feat_level,
+                 inputs_offsets,
+                 fpn_num_filters,
+                 apply_bn_for_resampling,
+                 conv_after_downsample,
+                 conv_bn_act_pattern,
+                 separable_conv,
+                 act_type,
+                 weight_method,
+                 data_format,
+                 name='fnode'):
+        super().__init__(name=name)
+        self.feat_level = feat_level
+        self.inputs_offsets = inputs_offsets
+        self.fpn_num_filters = fpn_num_filters
+        self.apply_bn_for_resampling = apply_bn_for_resampling
+        self.conv_after_downsample = conv_after_downsample
+        self.conv_bn_act_pattern = conv_bn_act_pattern
+        self.separable_conv = separable_conv
+        self.act_type = act_type
+        self.weight_method = weight_method
+        self.data_format = data_format
+        self.resample_layers = []
+        self.vars = []
```

Comment:
what is vars? variances? Please use explicit names.

Reply:
variables Octavio. They are the weights for combining the fpn nodes. I will rename them as variables.

Reply:
I renamed them as assign_weights as they are the scalar weights for fusing the features.